### PR TITLE
Unyson support reference #1237

### PR DESCRIPTION
**Pull request by meta4creations**
I would like to propose a couple new filters to add to the "extensions/shortcodes/includes/class-fw-shortcode.php" file. This would allow access to add global options and modified views of all, or limited amount, of existing shortcodes.

### CODE REVIEW

File: includes/class-fw-shortcode.php [hunk 1]
```diff
@@ -197,11 +197,11 @@ protected function _render($atts, $content = null, $tag = '')
 		}
 
 		$this->enqueue_static();
-		return fw_render_view($view_file, array(
+		return apply_filters('fw_shortcode_render_view', fw_render_view($view_file, array(
```
Comment (moldcraft) [inline includes/class-fw-shortcode.php]: Insted of filtering html (which has no sense and has overhead) maybe will be better to

```php
apply_filters('fw_shortcode_render_view_path', $view_file)
```

?

Comment (meta4creations) [inline includes/class-fw-shortcode.php]: The reason I kept the HTML passed through the filter was to keep the original view intact. My usage of this was to wrap the original view with another div and apply extra styles & attributes to the wrapper... I am not passing the HTML to parse and modify. My goal is to keep all the original views and add to them. If we only filtered the path, then I would need to recreate every view of every shortcode myself and then modify them... which, then would probably be easier to just copy the directories from the shortcode extension and modify that way.

I guess another alternative would be to add a before and after action around the current view render and then return all of that.

Thoughts?

Comment (moldcraft) [inline includes/class-fw-shortcode.php]: I will add an before and after feature.

Comment (moldcraft) [inline includes/class-fw-shortcode.php]: Done https://github.com/ThemeFuse/Unyson-Shortcodes-Extension/commit/e45fa01e8fa5730f9cc988e3c20a15cabf9bb86c

It's ok?

Comment (meta4creations) [inline includes/class-fw-shortcode.php]: Yes, I believe that will work! Thanks!

**Comment by meta4creations** [comment]
Do you know when an update to Unyson will be released that includes this update?

Thanks!

**Comment by moldcraft** [comment]
It will be an update of Shortcodes extension, not Unyson plugin. I will try this week.

**Comment by meta4creations** [comment]
Cool, thanks!

**Pull request merged**